template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/CmTemplateProcessor.java

```diff
@@ -298,10 +300,16 @@ public AutoscaleRecommendation recommendAutoscale() {
                 com.google.common.base.Optional<T> enumValue = Enums.getIfPresent(enumClass, service);
                 if (enumValue.isPresent()) {
                     if (EntitledForServiceScale.class.isAssignableFrom(enumClass)) {
-                        Entitlement entitledFor = ((EntitledForServiceScale) enumValue.get()).getEntitledFor();
+                        EntitledForServiceScale entitledForServiceScale = (EntitledForServiceScale) enumValue.get();
+                        Entitlement entitledFor = entitledForServiceScale.getEntitledFor();
                         if (!entitlements.contains(entitledFor.name())) {
-                            foundBlackListItem = true;
-                            break;
+                            if (isVersionEnablesScaling(version, entitledForServiceScale)) {
+                                foundBlackListItem = false;
+                                break;
+                            } else {
```

Comment:
1. `foundBlackListItem` is always `false` at this point. Any time it is set to `true` the loop is exited by a subsequent `break`.
2. I wonder if doing a `break` here is indeed the right thing, instead of either `continue` or simply not having any specific action (which is basically equivalent with `continue` as there are no other statements after the enclosing `if`). What if some `service1` is found okay due to `!entitlements.contains(entitledFor.name()) && isVersionEnablesScaling(version, entitledForServiceScale)` (and hence executing the statements inside this `if`), but some other `service2` occurring later (with respect to iteration order) in the `hostGroupServices.getValue()` collection would be blacklisted? With the current logic, the check for validity of `service2` is never made due to the `break`.